tests/testlib/s2n_hex_testlib.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <string.h>
+
+#include "error/s2n_errno.h"
+#include "stuffer/s2n_stuffer.h"
+#include "testlib/s2n_testlib.h"
+#include "utils/s2n_safety.h"
+
+S2N_RESULT s2n_stuffer_alloc_from_hex(struct s2n_stuffer *bytes_out, const char *hex_cstr)
+{
+    RESULT_ENSURE_REF(bytes_out);
+    RESULT_ENSURE_REF(hex_cstr);
+
+    DEFER_CLEANUP(struct s2n_blob hex = { 0 }, s2n_free);
+    /* Copying the hex into heap memory to handle the 'const' isn't exactly efficient,
+     * but for a testlib method it is sufficient.
+     */
+    RESULT_GUARD_POSIX(s2n_alloc(&hex, strlen(hex_cstr)));
+    RESULT_CHECKED_MEMCPY(hex.data, hex_cstr, hex.size);
+
+    RESULT_GUARD_POSIX(s2n_stuffer_alloc(bytes_out, strlen(hex_cstr) / 2));
+    RESULT_GUARD(s2n_stuffer_read_hex(bytes_out, &hex));
+    return S2N_RESULT_OK;
+}
+
+/* Unlike other hex methods, the hex string read here may include spaces.
```

Comment:
Nit: Move this comment to the header file so that my IDE picks it up whenever I mouse over the method?